src/Functions/FunctionsStringBytes.cpp

```diff
@@ -0,0 +1,142 @@
+#include <cmath>
+#include <Columns/ColumnConst.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnVector.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionsStringBytes.h>
+#include <Functions/IFunction.h>
+#include <Common/BitHelpers.h>
+#include <Common/PODArray.h>
+
+namespace DB
+{
+
+inline UInt8 countBits(UInt64 x)
+{
+#if defined(__clang__)
+    return __builtin_popcountll(x);
+#else
+    UInt8 count = 0;
+    while (x)
+    {
+        count += x & 1;
+        x >>= 1;
+    }
+    return count;
+#endif
+}
+
+class ByteCounters
+{
+private:
+    static constexpr size_t COUNTERS_SIZE = 256;
+    UInt32 counters[COUNTERS_SIZE] = {0};
+    UInt32 current_generation = 0;
+    UInt32 generation_mask = 0x80000000;
+    size_t total_count = 0;
+
+public:
+    void add(UInt8 byte)
+    {
+        UInt32 & counter = counters[byte];
+        if ((counter & generation_mask) != current_generation)
+            counter = current_generation;
+        counter++;
```

Comment:
A minor thing: we write prefix increment by default.

Reply:
Fixed.